### DISCUSSION TITLE
How can It run Https on  https://hoppscotch.io/?

### DISCUSSION THREAD

**Discussion by DavidHsu100** [hoppscotch/hoppscotch#1756] [Q&A] [answered]
Dear Sir：

How can It run Https on  https://hoppscotch.io/?
I want to set TLS cer and key to request https, but I don't know how to set it?

BR~

David

**Comment by liyasthomas** [comment]
This feature is not available at the moment. But surely is in our roadmap. Thanks for your patience, will update here once it's implemented.

**Reply by DavidHsu100** [reply]
Dear liyasthomas:

ok~ 
Thanks~

BR~

David

**Comment by DavidHsu100** [comment]
Dear liyasthomas:

I saw someone discuss about this, and you show some image as following:
wherer is the "No file" and "No cert" item?
Does Local development environment support this?

![image](https://user-images.githubusercontent.com/87112513/128118792-05373d3d-b92f-4bc5-8f41-0eb7cc97e50a.png)

BR~

David

**Comment by liyasthomas** [comment, marked as answer]
It seems you're using an outdated version of Hoppscotch project. Please use the latest version in Hoppscotch.io.

Still, certificate support is not available yet.

**Reply by DavidHsu100** [reply]
Dear liyasthomas:

If I want to add the https feature myself.
Where can I modify?

BR~

David

**Reply by liyasthomas** [reply]
Ah, what do you mean by `https feature`?
Please elaborate your exact use case.

**Reply by DavidHsu100** [reply]
Dear liyasthomas:

I add https to web server that is my device.  I  generate a certificate and setting up the TLS/ssl connection inside the web server. 
I need to configure the web client to handle the TLS/SSL. It need to set up the cert and key on client(hoppscotch), then it can do an internet connection secure.

BR~

David

**Comment by liyasthomas** [comment]
We use Axios HTTP client under the hood, read more about how Axios handle cert and key files in HTTP requests here: https://stackoverflow.com/questions/51363855/how-to-configure-axios-to-use-ssl-certificate

You can find the network strategies files under [this folder](https://github.com/hoppscotch/hoppscotch/tree/main/helpers/strategies). You'll also need a UI component that accept cert and key files.
